### PR TITLE
Fix meta encoding level

### PR DESCRIPTION
**Pull request by JorisGoosen**
This changes the way the `meta` is generated for options and makes sure it includes the levels and factors from RM ANOVA.

It also sends this meta to the engine for editImage and saveImage and makes sure that the levels shown there are decoded as well. There is a weird glitch though but Im not sure if that is my fault. 
The plot resizes, but always with a delay. Unless Don can see what is going there I suggest we make an issue after this is merged because with this PR at least the analysis works

### CODE REVIEW

File: Desktop/widgets/jasplistcontrol.h [hunk 1]
```diff
@@ -89,7 +89,7 @@ class JASPListControl : public JASPControl
 			const QString&		valueRole()					const			{ return _valueRole;			}
 			bool				containsVariables()			const			{ return _containsVariables;	}
 			bool				containsInteractions()		const			{ return _containsInteractions;	}
-			bool				encodeValue()				const override	{ return containsVariables();	}
+			bool				encodeValue()				const override	{ return containsVariables() || containsInteractions();	}
```
Comment (boutinb) [inline Desktop/widgets/jasplistcontrol.h:92]: Why did you add this?

**Review by boutinb — APPROVED** [review]

File: Desktop/widgets/jasplistcontrol.h (hunk 1 above)
Comment (JorisGoosen) [reply Desktop/widgets/jasplistcontrol.h:92]: Otherwise interactions between levels with encodable entries wouldn't be encoded. And thus RM ANOVA wouldnt work.

`withinModelTerms` specifically

Comment (boutinb) [reply Desktop/widgets/jasplistcontrol.h:92]: I would have then preferred to overwrite encodeValue() for the specific class...

Comment (JorisGoosen) [reply Desktop/widgets/jasplistcontrol.h:92]: Well, you wrote quite a lot of code, I thought this was the most specific place I could get.

But are there lists where the interactions don't need to be encoded?
